lib/samples/filter_features_in_scene/filter_features_in_scene.dart

```diff
@@ -0,0 +1,252 @@
+
+class FilterFeaturesInScene extends StatefulWidget {
+  const FilterFeaturesInScene({super.key});
+
+  @override
+  State<FilterFeaturesInScene> createState() => _FilterFeaturesInSceneState();
+}
+
+class _FilterFeaturesInSceneState extends State<FilterFeaturesInScene>
+    with SampleStateSupport {
+  // Create a controller for the scene view.
+  final _sceneViewController = ArcGISSceneView.createController();
+  // A flag for when the scene view is ready.
+  var _ready = false;
+
+  // ArcGIS Online services.
+  final _osmTopographic =
+      'https://www.arcgis.com/home/item.html?id=1e7d1784d1ef4b79ba6764d0bd6c3150';
+  final _elevationSource =
+      'https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer';
+  final _sanFranciscoBuildings =
+      'https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer';
+
+  late ArcGISSceneLayer _sfBuildingsSceneLayer;
+  late ArcGISSceneLayer _osmBuildingsSceneLayer;
+
+  late SceneLayerPolygonFilter _sceneLayerPolygonFilter;
+  // Graphic to get San Francisco's information.
+  Graphic _sfGraphic = Graphic();
+  // Graphics overlay to present the graphic for the sample.
+  final _sfGraphicsOverlay = GraphicsOverlay();
+
+  late Geometry _sceneLayerExtentPolygon;
+
+  // A state for filtering features in a scene.
+  SceneFilterAction _sceneFilterAction = SceneFilterAction.filter;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a scene view to the widget tree and set a controller.
+                  child: ArcGISSceneView(
+                    controllerProvider: () => _sceneViewController,
+                    onSceneViewReady: onSceneViewReady,
+                  ),
+                ),
+                Center(
+                  child: ElevatedButton(
+                    onPressed: onSceneActionPressed,
+                    child: Text(_sceneFilterAction.label),
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onSceneViewReady() async {
+    final scene = _setupScene();
+    _sceneViewController.arcGISScene = scene;
+
+    // Create a scene layer for the San Francisco buildings.
+    _sfBuildingsSceneLayer = ArcGISSceneLayer.withUri(
+      Uri.parse(_sanFranciscoBuildings),
+    );
+    await _sfBuildingsSceneLayer.load();
+
+    // Add the graphics overlay to the scene view controller.
+    _sceneViewController.graphicsOverlays.add(_sfGraphicsOverlay);
+
+    _createAndFilterPolygon();
+
+    addBuildings();
```

Comment:
Is there a reason one of these is private and the other not? Or can these be consistent?